keanu-project/src/main/java/io/improbable/keanu/util/csv/ColumnWriter.java

```diff
@@ -35,10 +37,11 @@ public File toFile(File file) {
                 List<Object> flatList = vertex.getValue().asFlatList();
                 row.add(i < flatList.size() ? flatList.get(i).toString() : getEmptyValue());
             }
-            String[] rowToString = new String[row.size()];
-            data.add(row.toArray(rowToString));
+            String[] rowArray = new String[row.size()];
+            writer.writeNext(row.toArray(rowArray), false);
         }
-        return writeToFile(file, data);
+        writer.close();
```

Comment:
`CSVWriter` implements `Closeable` which means you can use "try with resources"

https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
ditto in `RowWriter` and `SampleWriter`

Reply:
done